### PR TITLE
feat: enable AddSingleTypeSerializer to receive a type parameter

### LINKED ISSUE #201: AddSingleTypeResolver using Type
In order to be possible to create consumers/producers at runtime, the `AddSingleTypeResolver ` extension needs to work receiving a `Type` parameter in addition to the current behavior with `Generics`.

Current Behavior:
`.AddSingleTypeSerializer<PauseResumeMessage, ProtobufNetSerializer>()`

Desired behavior:
`.AddSingleTypeSerializer<ProtobufNetSerializer>(typeof(PauseResumeMessage))`
`.AddSingleTypeSerializer<ProtobufNetSerializer>(obj.GetType())`

### PR DESCRIPTION
**Pull request by dougolima**
# Description

Fixes #201

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            where TSerializer : class, ISerializer
+        {
+            middlewares.DependencyConfigurator.AddTransient<TSerializer>();
+
+            return middlewares.AddSerializer(
+                resolver => resolver.Resolve<TSerializer>(),
+                _ => new SingleMessageTypeResolver(messageType));
+        }
+    }
+}
```
Comment (filipeesch) [inline src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs:138]: Missing an overload where we can pass the Type instance and a serializer Factory in Consumer and Producer builder

File: src/KafkaFlow.Serializer/ConsumerConfigurationBuilderExtensions.cs [hunk 2]
```diff
@@ -1,11 +1,12 @@
 ﻿namespace KafkaFlow
 {
+    using System;
     using KafkaFlow.Configuration;
 
     /// <summary>
-    /// Extension methods for <see cref="IConsumerMiddlewareConfigurationBuilder"/> and <see cref="IProducerMiddlewareConfigurationBuilder"/>
+    /// Extension methods for <see cref="IConsumerMiddlewareConfigurationBuilder"/>
```
Comment (filipeesch) [inline src/KafkaFlow.Serializer/ConsumerConfigurationBuilderExtensions.cs:7]: I would change this to "no needed"

File: src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
+﻿namespace KafkaFlow
+{
+    using System;
+    using KafkaFlow.Configuration;
+
+    /// <summary>
+    /// Extension methods for <see cref="IProducerMiddlewareConfigurationBuilder"/>
```
Comment (filipeesch) [inline src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs:7]: I would change this to "no needed"

**Review by filipeesch — CHANGES_REQUESTED** [review]

File: src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs (hunk 3 above)
Comment (dougolima) [reply src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs]: done

File: src/KafkaFlow.Serializer/ConsumerConfigurationBuilderExtensions.cs (hunk 2 above)
Comment (dougolima) [reply src/KafkaFlow.Serializer/ConsumerConfigurationBuilderExtensions.cs]: done

File: src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs (hunk 1 above)
Comment (dougolima) [reply src/KafkaFlow.Serializer/ProducerConfigurationBuilderExtensions.cs:157]: Sure, added it

**Review by filipeesch — APPROVED** [review]

**Pull request merged**